GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -285,7 +270,15 @@ public override bool TryDownloadNewestVersion(out string errorMessage)
             {
                 try
                 {
-                    this.DownloadedPackagePath = this.nuGetFeed.DownloadPackageAsync(this.highestVersionAvailable.Identity).GetAwaiter().GetResult();
+                    PackageIdentity packageId = this.GetPackageForVersion(this.highestVersionAvailable);
+
+                    if (packageId == null)
+                    {
+                        errorMessage = "Could not find package for version. This indicates the package feed is out of sync.";

```

Comment:
> the package feed is out of sync

Out of sync with what?

Reply:
I will clarifying this message... but, it means that the version it has been instructed to download does not exist in the feed (e.g., the version returned by the `orgInfoServer` is out of sync with what is available in the feed)